compiler/dyno/include/chpl/resolution/resolution-types.h

```diff
@@ -29,6 +29,7 @@
 #include "chpl/uast/Function.h"
 #include "chpl/util/bitmap.h"
 #include "chpl/util/memory.h"
+#include "chpl/uast/all-uast.h"
```

Comment:
Can we forward declare instead? Like



```
namespace chpl {
  namespace uast {
    class For;
  }
}
```



That's the strategy I've been trying to use elsewhere, just to try to keep the amount of C++ headers required (and compile times) down.